subsys/net/lib/lwm2m_client_utils/Kconfig

```diff
@@ -7,6 +7,10 @@
 menuconfig LWM2M_CLIENT_UTILS
 	bool "LWM2M client utilities library"
 	select LWM2M if !ZTEST
+	imply LTE_LC_EDRX
+	imply LTE_LC_NEIGHBOR_CELL_MEAS
+	imply LTE_LC_PSM
+	imply LTE_LC_TAU_PRE_WARNING
```

Comment:
This was discussed extensively with colleagues and we came to the conclusion that this is better to prevent circular dependencies.

Reply:
Library users shouldn't need to define the dependencies what the library needs but a given library should handle them by itself. That being said, I don't know if the Kconfig system is just clumsy enough that `depends on` is the only possibility in some cases. Here imply seems to work fine so far so I'd prefer to keep it.

Reply:
If the code compiles without this, then imply is fine.
But if the build breaks without it, then I would use `depends on`.

Also, `depends on` hides the section from the menuconfig, if the dependencies are not enabled.

https://docs.zephyrproject.org/latest/build/kconfig/tips.html#the-imply-statement

Reply:
Unfortunately, in those places where I used `imply` it's necessary because otherwise it fails immediately at compliance due to dependency loops.

Reply:
and this is often because other parts in Kconfig has been using an improper `select`.

The best thing to do when dealing with prompt-based symbols is `depends on`, for a lot of reasons.
One of them is that it's possible downstream to adjust dependencies and thereby provide alternative implementations for a given feature. But if someone upstream uses `select` (and to some extent `imply`), then it is not possible to get rid of such selection.

some additional guideline: https://www.kernel.org/doc/html/v6.9/kbuild/kconfig-language.html
> **Note:**
> select should be used with care. select will force a symbol to a value without visiting the dependencies. By abusing select you are able to select a symbol FOO even if FOO depends on BAR that is not set. In general use select only for non-visible symbols (no prompts anywhere) and for symbols with no dependencies. That will limit the usefulness but on the other hand avoid the illegal configurations all over.